AutoNode/monitor.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import time
+import json
+import datetime
+import traceback
+
+from pyhmy import (
+    cli,
+    json_load,
+    Typgpy
+)
+
+from .common import (
+    harmony_dir,
+    validator_config,
+    node_config,
+    save_node_config
+)
+from .node import (
+    wait_for_node_response,
+    assert_no_bad_blocks,
+    check_and_activate
+)
+from .blockchain import (
+    get_latest_header,
+    get_latest_headers,
+    get_all_validator_addresses,
+    get_validator_information,
+    get_sharding_structure,
+)
+from .util import (
+    can_check_blockchain,
+)
+
+log_path = f"{harmony_dir}/autonode_monitor.log"
+
+_check_interval = 8  # Estimated block time
+
+
+def _run_monitor(shard_endpoint):
+    start_time = time.time()
+    wait_for_node_response(node_config['endpoint'], tries=900, sleep=1)  # Try for 15 min
+    wait_for_node_response(shard_endpoint, tries=900, sleep=1)  # Try for 15 min
+    wait_for_node_response("http://localhost:9500/", verbose=True)
+    count = 0
+    while get_latest_header('http://localhost:9500/')['blockNumber'] == 0:
```

Comment:
could have been a cool usage of python3 async/await

Reply:
oh right, might do in the future :)